gateway/src/apicast/policy/tls/tls.lua

```diff
@@ -154,10 +152,45 @@ end
 -- @tparam[opt] table config Policy configuration.
 function _M.new(config)
   local self = new(config)
-
   self.certificates = parse_config_certificates(config and config.certificates or empty)
 
   return self
 end
 
+-- Set the given certificate to be the default one for the request.
+-- @tparam cert: certificate table with certificates in der format
+local function set_certificate(cert)
+  local ok, err = ssl.set_cert(cert.certificate)
+  if not ok then
+    ngx.log(ngx.ERR, "failed to set certificate: ", err)
+    return err
+  end
+
+  local ok, err = ssl.set_priv_key(cert.certificate_key)
+  if not ok then
+    ngx.log(ngx.ERR, "failed to set DER private key: ", err)
+    return err
+  end
+end
+
+function _M:ssl_certificate()
+  if #self.certificates == 0 then
+    ngx.log(ngx.WARN, "No valid certificates loaded")
+    return
+  end
+
+  local ok, err = ssl.clear_certs()
+  if not ok then
+      ngx.log(ngx.ERR, "failed to clear existing (fallback) certificates")
+      return ngx.exit(ngx.ERROR)
+  end
+
+  for _, cert in ipairs(self.certificates) do
+    if not set_certificate(cert) then
+      -- Certificate is set correctly, use this one and end the loop.
```

Comment:
Shouldn't this terminate the ssl_certificate phase then?

Reply:
I'm not sure If I get this one, just abort the rest of ssl_certificate policies?

Reply:
What is the desired outcome if you have this policy twice? Or some that tries to do the same. How they would know it is already set? You can terminate the phase by ngx.exit(ngx.OK)